lib/solarwinds_apm/support/otlp_endpoint.rb

```diff
@@ -0,0 +1,196 @@
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+
+module SolarWindsAPM
+  # OTLPEndPoint
+  class OTLPEndPoint
+    SW_ENDPOINT_REGEX = /^apm\.collector\.([a-z0-9-]+)\.cloud\.solarwinds\.com$/
+    OTEL_ENDPOINT_REGEX = %r{^https://otel\.collector\.([a-z0-9-]+)\.cloud\.solarwinds\.com:443(?:/.*)?$}
+    OTEL_ENDPOINT_LOCAL_REGEX = %r{\Ahttp://0\.0\.0\.0:(4317|4318)\z}
+    OTEL_ENDPOINT_LOCAL_REGEX2 = %r{\Ahttp://0\.0\.0\.0:(4317|4318)/v1/(metrics|traces|logs)\z}
+    DEFAULT_OTLP_ENDPOINT = 'https://otel.collector.na-01.cloud.solarwinds.com:443'
+    DEFAULT_APMPROTO_ENDPOINT = 'apm.collector.na-01.cloud.solarwinds.com'
+
+    def initialize
+      @token = nil
+      @service_name = nil
+      @lambda_env = determine_lambda_env
+      @agent_enable = true
+      @localhost = false
+      determine_if_localhost
+    end
+
+    def determine_if_localhost
+      @localhost = true if ENV['OTEL_EXPORTER_OTLP_ENDPOINT'].to_s.match?(OTEL_ENDPOINT_LOCAL_REGEX)
+      @localhost = true if ENV['OTEL_EXPORTER_OTLP_TRACES_ENDPOINT'].to_s.match?(OTEL_ENDPOINT_LOCAL_REGEX2)
+      @localhost = true if ENV['OTEL_EXPORTER_OTLP_METRICS_ENDPOINT'].to_s.match?(OTEL_ENDPOINT_LOCAL_REGEX2)
+      @localhost = true if ENV['OTEL_EXPORTER_OTLP_LOGS_ENDPOINT'].to_s.match?(OTEL_ENDPOINT_LOCAL_REGEX2)
+    end
+
+    def config_otlp_endpoint
+      config_service_name
+      config_token
+      %w[TRACES METRICS LOGS].each { |data_type| configure_otlp_endpoint(data_type) }
+    end
+
+    def config_service_name
+      resource_attributes = ENV['OTEL_RESOURCE_ATTRIBUTES'].to_s.split(',').each_with_object({}) do |resource, hash|
+        key, value = resource.split('=')
+        hash[key] = value
+      end
+
+      @service_name = if @lambda_env
+                        ENV['OTEL_SERVICE_NAME'] || ENV['AWS_LAMBDA_FUNCTION_NAME'] || resource_attributes['service.name']
+                      else
+                        ENV['OTEL_SERVICE_NAME'] || resource_attributes['service.name'] || @service_name || 'None'
+                      end
+
+      ENV['OTEL_SERVICE_NAME'] = @service_name
+    end
+
+    def mask_token(token)
+      token = token.to_s
+      return '*' * token.length if token.length <= 4
+
+      "#{token[0, 2]}#{'*' * (token.length - 4)}#{token[-2, 2]}"
+    end
+
+    def config_token(data_type)
+      agent_enable = true
+
+      if @localhost
+        # for localhost, only valid SW_APM_SERVICE_KEY can make agent enable
+        @agent_enable = false unless valid?(ENV['SW_APM_SERVICE_KEY'])
+        return
+      end
```

Comment:
i'm not sure if this OTLPEndPoint module is intended to be shared by inside and outside lambda--if it's only meant for outside lambda, this above logic needs to be removed. if it will be used inside lambda, it should be revised to an `is_lambda?` check (maybe one already exists) rather than testing for localhost.

Reply:
It will be used inside lambda, I added check if both localhost and lambda_env, return false